velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -640,6 +646,58 @@ class S3FileSystem::Impl {
     return getDefaultCredentialsProvider();
   }
 
+  // Return a client RetryStrategy based on the config.
+  std::optional<std::shared_ptr<Aws::Client::RetryStrategy>> getRetryStrategy()
+      const {
+    auto retryMode = hiveConfig_->s3RetryMode();
+    auto maxAttempts = hiveConfig_->s3MaxAttempts();
+    if (retryMode.has_value()) {
```

Comment:
If user doesn't config retryMode but config maxAttempts, does maxAttempts take effect? It should takes effect since we have default retry mode.

Reply:
No. In velox, `maxAttempts` need work together with `retryMode`. If `retryMode` isn't configured, S3 client will be created w/o `RetryStrategy`. But for Gluten, we've set default value for `retryMode`. I updated the doc in this PR.